src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,75 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SEPARATOR = ",";
+    private static final String NEW_LINE = "\n";
+    private static final String LABEL_SUPPLY = "supply";
+    private static final String LABEL_BUY = "buy";
+    private static final String LABEL_RESULT = "result";
+    private static final int VALUE_INDEX = 1;
+    private static final int LABEL_INDEX = 0;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String fileContent = readFile(fromFileName);
+        String report = generateReport(fileContent);
+        writeToFile(report, toFileName);
+    }
+
+    private String readFile(String fileName) {
+        StringBuilder content = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                content.append(line).append(NEW_LINE);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Error while reading data from the file ", e);
+        }
+        return content.toString();
+    }
+
+    private String generateReport(String fileContent) {
+        int supplyTotal = 0;
+        int buyTotal = 0;
+
+        StringBuilder reportBuilder = new StringBuilder();
+
+        String[] lines = fileContent.split(NEW_LINE);
+        for (String line : lines) {
+            String[] parts = line.split(SEPARATOR);
+            if (parts.length == 2) {
+                String label = parts[LABEL_INDEX];
+                int value = Integer.parseInt(parts[VALUE_INDEX]);
+                if (LABEL_SUPPLY.equals(label)) {
+                    supplyTotal += value;
+                } else if (LABEL_BUY.equals(label)) {
+                    buyTotal += value;
+                }
+            }
+        }
```

Comment:
I think you can divide logic for one more private helpful method

Reply:
I don't know how to do this corectly and also i don't think it affects my code somehow